src/main/java/com/addthis/meshy/Meshy.java

```diff
@@ -200,7 +202,7 @@ public static InputStream getInput(int length, ByteBuf buffer) {
     }
 
     @Override public void close() {
-        closeAsync().syncUninterruptibly();
+        closeAsync().awaitUninterruptibly().syncUninterruptibly();
```

Comment:
Not sure why we need both of these. It's possible that we are trying to discard exceptions during shutdown. I don't think we also need syncUninterruptibly if await is being called.